overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +77,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return defaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	var snapshots map[uint64]*snapshotState
```

Comment:
is it worth using a two phase approach like for "snaps", with *json.RawMessage ?

Reply:
Yes. Done. Thanks for the idea.